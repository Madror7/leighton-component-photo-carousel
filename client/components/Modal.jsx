import React from 'react';
import { ModalContainer,
  Content,
  ModalCounter,
  ModalBar,
  ModalTabLink,
  ModalGalleryActions,
  ModalFooter,
  PropertyDetails,
  GalleryActionItem,
  ItemDiv,
  GalleryButton,
  ModalButton,
} from '../styles';

class Modal extends React.Component {
  constructor(props) {
    super(props);

    this.closeModal = this.closeModal.bind(this);
  }

  closeModal(event) {
    const { closeModal } = this.props;
    if (event.target.className.includes('can-close') || event.target.innerText === 'close') {
      closeModal();
    }
  }

  render() {
    const { display, btnBack, btnNext, id, imageCount, link, details } = this.props;
    return (
      <ModalContainer className="can-close" style={{display}} onClick={this.closeModal}>
        <ModalBar>
          <ModalTabLink>
            <div>Photos</div>
          </ModalTabLink>
          <ModalGalleryActions>
            <GalleryActionItem>
              <ItemDiv>
                <GalleryButton>Contact agent</GalleryButton>
              </ItemDiv>
            </GalleryActionItem>
            <GalleryActionItem>
              <ItemDiv>
                <i className="material-icons icon-light md-18">favorite_border</i>
                <span>Save Home</span>
              </ItemDiv>
            </GalleryActionItem>
            <GalleryActionItem>
              <ItemDiv>
                <i className="material-icons icon-light md-18">email</i>
                <span>Share</span>
              </ItemDiv>
            </GalleryActionItem>
            <GalleryActionItem>
              <ItemDiv id="close" onClick={this.closeModal}>
                <i className="material-icons icon-light md-30">close</i>
              </ItemDiv>
            </GalleryActionItem>
          </ModalGalleryActions>
        </ModalBar>
        <ModalButton id="back" onClick={() => { btnBack(id) }}><i className="material-icons md-36 icon-light back-icon">arrow_back_ios</i></ModalButton>
        <Content>
          <ModalCounter>
            {id + 1}
            {' of '}
            {imageCount}
          </ModalCounter>
          <img src={link} alt="fullsize version" />
        </Content>
        <ModalButton id="forward" onClick={() => { btnNext(id) }}><i className="material-icons md-36 icon-light">arrow_forward_ios</i></ModalButton>
        <ModalFooter>
          <PropertyDetails>
            {'For Sale: $'}
            {details.price}
            {' ('}
            {details.bed_count}
            {' beds, '}
            {details.bath_count}
            {' baths, '}
            {details.sq_ft}
            {' sqft)'}
          </PropertyDetails>
        </ModalFooter>
      </ModalContainer>
    );
  }
}

export default Modal;
