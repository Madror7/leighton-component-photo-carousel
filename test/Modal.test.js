import React from 'react';
import { shallow } from 'enzyme';

import Modal from '../client/components/Modal';

describe('Modal', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Modal details={{}} debug />);

    expect(component).toMatchSnapshot();
  });
});
