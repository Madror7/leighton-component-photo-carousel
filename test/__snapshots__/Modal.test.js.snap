// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly in "debug" mode 1`] = `
<styled.div
  className="can-close"
  onClick={[Function]}
  style={
    Object {
      "display": undefined,
    }
  }
>
  <styled.div>
    <styled.div>
      <div>
        Photos
      </div>
    </styled.div>
    <styled.ul>
      <styled.li>
        <styled.div>
          <styled.button>
            Contact agent
          </styled.button>
        </styled.div>
      </styled.li>
      <styled.li>
        <styled.div>
          <i
            className="material-icons icon-light md-18"
          >
            favorite_border
          </i>
          <span>
            Save Home
          </span>
        </styled.div>
      </styled.li>
      <styled.li>
        <styled.div>
          <i
            className="material-icons icon-light md-18"
          >
            email
          </i>
          <span>
            Share
          </span>
        </styled.div>
      </styled.li>
      <styled.li>
        <styled.div
          onClick={[Function]}
        >
          <i
            className="material-icons icon-light md-30"
          >
            close
          </i>
        </styled.div>
      </styled.li>
    </styled.ul>
  </styled.div>
  <styled.button
    className="back"
    onClick={[Function]}
  >
    <i
      className="material-icons md-36 icon-light"
    >
      arrow_back_ios
    </i>
  </styled.button>
  <styled.div>
    <styled.div>
      NaN
       of 
    </styled.div>
    <img
      alt="fullsize version"
      className="imgs"
    />
  </styled.div>
  <styled.button
    className="forward"
    onClick={[Function]}
  >
    <i
      className="material-icons md-36 icon-light"
    >
      arrow_forward_ios
    </i>
  </styled.button>
  <styled.div>
    <styled.div>
      For Sale: $
       (
       beds, 
       baths, 
       sqft)
    </styled.div>
  </styled.div>
</styled.div>
`;
